### DISCUSSION TITLE
How can I exchange a callback response for a different token?

### DISCUSSION THREAD

**Discussion by Nfinished** [nextauthjs/next-auth#3438] [Help] [answered]
I've got a login flow such that after google/facebook/whatever calls back with a profile and token, I need to forward that response to a service that will then return a user object and JWT needed for the session. I've got a custom initializer but can't seem to find the user data returned to /callback, is there another way I should be going about this?

**Comment by nik32** [comment]
@Nfinished have you tried this [Add auth support to a Next.js app with a custom backend](https://arunoda.me/blog/add-auth-support-to-a-next-js-app-with-a-custom-backend).
NOTE - you can get accessToken by following the below links - 
[reactjs - how to get the provider access token in next-auth - Stack Overflow](https://stackoverflow.com/questions/69068495/how-to-get-the-provider-access-token-in-next-auth#:~:text=Please%20be%20aware%20of%20next,wih%20callbacks%20in%20next0auth%20V4.) [Answer by  - [Safwat Fathi](https://stackoverflow.com/users/7424232/safwat-fathi)] 
[Accessing the Google Access Token in NextAuth.js | Srijan's Blog](https://blog.srij.dev/nextauth-google-access-token)

**Comment by Nfinished** [comment, marked as answer]
Thanks, I ended up finding the support I needed via the `token`option. https://next-auth.js.org/configuration/providers/oauth#token-option
